gradle/build.versions.toml

```diff
@@ -0,0 +1,42 @@
+[versions]
+asm = "9.3"
+jackson = "2.13.2"
+junit5 = "5.7.1"
+spock = "2.1-groovy-3.0"
+
+[libraries]
+ant = "org.apache.ant:ant:1.10.8"
+apache-compress = "org.apache.commons:commons-compress:1.21"
+apache-rat = "org.apache.rat:apache-rat:0.11"
+asm = { group = "org.ow2.asm", name="asm", version.ref="asm" }
+asm-tree = { group = "org.ow2.asm", name="asm-tree", version.ref="asm" }
+bytebuddy = "net.bytebuddy:byte-buddy:1.12.9"
+checkstyle = "com.puppycrawl.tools:checkstyle:9.3"
+commons-codec = "commons-codec:commons-codec:1.11"
+commmons-io = "commons-io:commons-io:2.2"
+docker-compose = "com.avast.gradle:gradle-docker-compose-plugin:0.14.13"
+forbiddenApis = "de.thetaphi:forbiddenapis:3.2"
+hamcrest = "org.hamcrest:hamcrest:2.1"
+httpcore = "org.apache.httpcomponents:httpcore:4.4.12"
+httpclient = "org.apache.httpcomponents:httpclient:4.5.10"
+idea-ext = "gradle.plugin.org.jetbrains.gradle.plugin.idea-ext:gradle-idea-ext:1.1.1"
+json-schema-validator = "com.networknt:json-schema-validator:1.0.49"
+jackson-dataformat-yaml = { group = "com.fasterxml.jackson.dataformat", name="jackson-dataformat-yaml", version.ref="jackson" }
+jackson-platform = { group = "com.fasterxml.jackson", name="jackson-bom", version.ref="jackson" }
+jna = "net.java.dev.jna:jna:5.10.0"
+junit = "junit:junit:4.12"
+junit5-platform = { group = "org.junit", name="junit-bom", version.ref="junit5" }
```

Comment:
If we're using the platform bom shouldn't we just omit version numbers from the dependencies below? Also they don't seem to match. We're using version 5.7.1 of the BOM but 5.8.1 of the individual dependencies.

Reply:
there's is gap in version catalogue support that does not let allow you to declare a dependency without a version. therefore we I needed to add the version there too. I think that's fine as the interesting bit of platforms is the handling of transitive deps their versions.

Interestingly I took the dependencies that are 5.8.1 as they we're actually used in our build already. turns out the Spock dependency brings in a newer version of the `junit5-bom`. 🤯 I've upgraded version we use to 5.8.1 but referenced the junit5 version used in the other dependency declarations with that junit5 version.